plugin/trino-hive/src/main/java/io/trino/plugin/hive/ion/IonFileWriterFactory.java

```diff
@@ -100,17 +99,19 @@ public Optional<FileWriter> createFileWriter(
                     .mapToObj(ordinal -> new Column(fileColumnNames.get(ordinal), fileColumnTypes.get(ordinal), ordinal))
                     .toList();
 
-            IonWriterBuilder writerBuilder = getIonEncoding(schema).equals(TEXT_ENCODING)
-                    ? IonTextWriterBuilder.minimal()
-                    : IonBinaryWriterBuilder.standard();
+            Boolean binaryEncoding = switch (getIonEncoding(schema).toLowerCase(Locale.ROOT)) {
+                case TEXT_ENCODING -> true;
+                case BINARY_ENCODING -> false;
+                default -> throw new IllegalArgumentException("Unsupported Ion encoding format");
```

Comment:
Maybe we should use the `TrinoException` here.
```suggestion
                default -> throw new TrinoException(HIVE_UNSUPPORTED_FORMAT, "Unsupported Ion encoding format");
```

Reply:
Will update this in the next commit. Thanks